.github/workflows/build.yml

```
@@ -0,0 +1,66 @@
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
```

Set least privileges to the GH token:

```suggestion

permissions:
  contents: read
```

---

I think we would also need `security-events: write` for the sarif upload.